### PR TITLE
Add Dockerfile to build for Debian

### PR DESCRIPTION
**Pull request by jman-schief**
This will add a Dockefile to build a .deb for Debian.

It works but I'll iterate a bit more on this to see if I can slim this down a bit

(also suggestions and reviews welcome)

Ref: #235

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -1,3 +1,28 @@
+# Build emacs-ng inside a Docker container
```
Comment (brotzeit) [inline docker/README.md:1]: I think we should put the docs to the readme folder.

Comment (jman-schief) [reply docker/README.md:1]: you mean moving this part into `./docs` (the web documntation), correct?

(I agree)

File: mkdocs.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ nav:
     - Overview: README.md
     - Setup:
       - Build from source: building.md
+      - Build with Docker: docker.md
```
Comment (jman-schief) [inline mkdocs.yml:13]: @brotzeit moved the relevant documentation here, is that ok?

**Comment by jman-schief** [comment]
@brotzeit minus comments or corrections, I think we can merge this

**Review by black7375 — APPROVED** [review]

**Comment by black7375** [comment]
Can we distribute image through [github docker registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry) or docker hub?

The disadvantage of the github registry is that [login is required](https://github.community/t/docker-pull-from-public-github-package-registry-fail-with-no-basic-auth-credentials-error/16358/87)  when downloading.

File: docs/docker.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Build emacs-ng using a Docker container
+
+To ease the job of building emacs-ng, we are preparing Docker images
+to allow a fully containerized build without installing any build dependencies.
+
+Bulding this will create a Docker image of about ~9gb and will take
+quite some time (~45h, depending on your workstation specs). Once the
```
Comment (kidd) [inline docs/docker.md:7]: I hope it's 45m.... :)

**Comment by jman-schief** [comment]
> Can we distribute image through [github docker registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry) or docker hub?

That would spare the user the time to build the image, but you will end up anyway with a 9gb (!) image locally. This Dockerfile is to ease the job of building emacs-ng, not to distribute it. For the purpose of distributing it, I would see more suitable a channel releasing binary packages (.deb, .rpm, .aur, ...), example a continuous delivery service with a github workflow.

**Comment by black7375** [comment]
Ok, I understand.

**Review by DavidDeSimone — APPROVED** [review]
LGTM

File: mkdocs.yml (hunk 2 above)
Comment (brotzeit) [reply mkdocs.yml:13]: Yeah that's fine.

**Pull request merged**